shell/platform/android/test/io/flutter/view/AccessibilityBridgeTest.java

```diff
@@ -1827,6 +1831,59 @@ public void releaseDropsChannelMessageHandler() {
     verify(mockChannel, never()).setAccessibilityFeatures(anyInt());
   }
 
+  @Test
+  public void sendFocusAccessibilityEvent() {
+    AccessibilityManager mockManager = mock(AccessibilityManager.class);
+    AccessibilityChannel accessibilityChannel =
+        new AccessibilityChannel(mock(DartExecutor.class), mock(FlutterJNI.class));
+
+    ContentResolver mockContentResolver = mock(ContentResolver.class);
+    View mockRootView = mock(View.class);
+    Context context = mock(Context.class);
+    when(mockRootView.getContext()).thenReturn(context);
+    when(context.getPackageName()).thenReturn("test");
+    ViewParent mockParent = mock(ViewParent.class);
+    when(mockRootView.getParent()).thenReturn(mockParent);
+    when(mockManager.isEnabled()).thenReturn(true);
+
+    AccessibilityBridge accessibilityBridge =
+        setUpBridge(mockRootView, accessibilityChannel, mockManager, null, null, null);
+
+    HashMap<String, Object> arguments = new HashMap<>();
+    arguments.put("type", "focus");
+    arguments.put("nodeId", 123);
+    BasicMessageChannel.Reply reply = mock(BasicMessageChannel.Reply.class);
+    accessibilityChannel.parsingMessageHandler.onMessage(arguments, reply);
+
+    // Check that focus event was sent.
+    ArgumentCaptor<AccessibilityEvent> eventCaptor =
+        ArgumentCaptor.forClass(AccessibilityEvent.class);
+    verify(mockParent).requestSendAccessibilityEvent(eq(mockRootView), eventCaptor.capture());
+    AccessibilityEvent event = eventCaptor.getAllValues().get(0);
+    assertEquals(event.getEventType(), AccessibilityEvent.TYPE_VIEW_FOCUSED);
+    assertEquals(event.getSource(), null);
+  }
+
+  @Test
+  public void SetSourceAndPackageNameForAccessibilityEvent() {
+    AccessibilityManager mockManager = mock(AccessibilityManager.class);
+    ContentResolver mockContentResolver = mock(ContentResolver.class);
+    View mockRootView = mock(View.class);
+    Context context = mock(Context.class);
+    when(mockRootView.getContext()).thenReturn(context);
+    when(context.getPackageName()).thenReturn("test");
+    when(mockManager.isEnabled()).thenReturn(true);
+
+    AccessibilityBridge accessibilityBridge =
+        setUpBridge(mockRootView, null, mockManager, null, null, null);
+
+    AccessibilityEvent mockEvent = mock(AccessibilityEvent.class);
+    accessibilityBridge.obtainAccessibilityEvent(
```

Comment:
I think we want to test accessibilityBridge.onFocus.

You can do that by doing something like

```java
    AccessibilityBridge spyAccessibilityBridge = spy(accessibilityBridge);
    AccessibilityEvent mockEvent = mock(AccessibilityEvent.class);

    when(spyAccessibilityBridge.obtainAccessibilityEvent(AccessibilityEvent.TYPE_VIEW_FOCUSED))
        .thenReturn(mockEvent);

accessibilityBridge.onFocus(123);

verify(mockEvent).setSource(eq(mockRootView), eq(123));
    verify(mockEvent).setPackageName("test");
  }

```

Reply:
We can't do accessibilityBridge.onFocus because onFocus is a mothod in AccessibilityMessageHandler, we can only trigger it in accessibilityBridge by  accessibilityChannel.parsingMessageHandler.onMessage(arguments, reply), but in that way we can't spy it.

Reply:
Ah I keep thinking it was a method in a11y bridge, can you add visible to test and make accessibilityMessageHandler public?

Reply:
Even if I make accessibilityMessageHandler public. I don't know how to hook it up with a spybridge. Because accessibilityMessageHandler is set when creating a bridge, but spy come after.